feat: add C implementation for `blas/base/dsyr`

ShabiShett07 opened the pull request:
Progresses #2039

## Description

> What is the purpose of this pull request?

This pull request:

-   adds package `blas/base/dsyr`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2039 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add c implementation for blas/base/dsyr

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
feat: add c implementation for blas/base/dsyr

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
fix: file permissions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
remove: floating point

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: minor changes

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: change datatypes

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update implementation

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update jsdoc

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update datatypes

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add indentation

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
remove: whitespace

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, this needs review from you

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
c

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up\

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add checks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
Merge branch 'develop' into feature/dsyr

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
/stdlib merge

ShabiShett07 commented:
@kgryte, I have tried to replicate the changes that you made in `dgemv` in this PR.

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr.c:
```diff
@@ -0,0 +1,76 @@
+* @param layout    storage layout
+* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
+* @param N        number of elements along each dimension of `A`
+* @param alpha    scalar
```
```suggestion
* @param alpha    scalar constant
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_cblas.c:
```diff
@@ -0,0 +1,37 @@
+* @param layout    storage layout
+* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
+* @param N        number of elements along each dimension of `A`
+* @param alpha    scalar
```
```suggestion
* @param alpha    scalar constant
```

ShabiShett07 commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_ndarray.c:
```diff
@@ -0,0 +1,118 @@
+*
+* @param uplo      specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
+* @param N         number of elements along each dimension of `A`
+* @param alpha     scalar
```
```suggestion
* @param alpha     scalar constant
```

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: minor clean-up

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_ndarray.c:
```diff
@@ -0,0 +1,118 @@
+
+	// Perform input argument validation...
+	if ( uplo != CblasLower && uplo != CblasUpper ) {
+		c_xerbla( 1, "c_dsyr_ndarray", "Error: invalid argument. First argument must specify whether to reference the lower or upper triangular matrixecond argument must specify whether to reference the lower or upper triangular matrix. Value: `%d`.", uplo );
```
The error message needs updating.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr.c:
```diff
@@ -0,0 +1,83 @@
+		return;
+	}
+	if ( uplo != CblasLower && uplo != CblasUpper ) {
+		c_xerbla( 2, "c_dsyr", "Error: invalid argument. Second argument must specify whether to reference the lower or upper triangular matrixecond argument must specify whether to reference the lower or upper triangular matrix. Value: `%d`.", uplo );
```
The error message needs updating.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/lib/dsyr.native.js:
```diff
@@ -0,0 +1,88 @@
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param {string} order - storage layout
+* @param {string} uplo - specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
```
Use the description in `dsyr.js`.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/lib/ndarray.native.js:
```diff
@@ -0,0 +1,79 @@
+/**
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param {string} uplo - specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
```
Use the description in `ndarray.js`.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/lib/ndarray.native.js:
```diff
@@ -0,0 +1,79 @@
+*
+* @param {string} uplo - specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
+* @param {NonNegativeInteger} N - number of elements along each dimension of `A`
+* @param {number} alpha - scalar
```
```suggestion
* @param {number} alpha - scalar constant
```

Applies elsewhere, as well.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/examples/c/example.c:
```diff
@@ -0,0 +1,52 @@
+int main( void ) {
+	// Define a 3x3 symmetric matrix stored in row-major layout:
+	double A[ 3*3 ] = {
+		1.0, 0.0, 0.0,
```
This matrix is not symmetric.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/docs/repl.txt:
```diff
@@ -46,11 +46,25 @@
     --------
+    // Standard usage:
     > var x = new {{alias:@stdlib/array/float64}}( [ 1.0, 1.0 ] );
     > var A = new {{alias:@stdlib/array/float64}}( [ 1.0, 2.0, 0.0, 2.0 ] );
```
We should update these examples such that `A` is actually a 2x2 symmetric matrix, rather than being assymetric.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -219,7 +263,40 @@ TODO
+int main( void ) {
+    // Define a 3x3 symmetric matrix stored in row-major layout:
+    double A[ 3*3 ] = {
+        1.0, 0.0, 0.0,
```
Same comment regarding symmetry.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -219,7 +263,40 @@ TODO
+    }
+
+    // Perform the symmetric rank 1 operation `A = α*x*x^T + A` using alternative semantics indexing:
+    c_dsyr_ndarray( CblasUpper, N, 1.0, x, 1, 0, A, N, 1, 0 );
```
You need to allocate a separate `A` matrix before calling this function, as `A` is no longer symmetric given L285.

kgryte replied:
You can name them `A1` and `A2`.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
````diff
@@ -183,21 +183,65 @@ console.log( A );
+```c
+#include "stdlib/blas/base/shared.h"
+
+double A[] = { 1.0, 2.0, 3.0, 0.0, 1.0, 2.0, 0.0, 0.0, 1.0 };
````
`A` should be symmetric. We should update all examples accordingly. Otherwise, the lack of symmetry is confusing given the function descriptions.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -47,11 +47,11 @@ dsyr( 'row-major', 'upper', 3, 1.0, x, 1, A, 3 );
 
 -   **order**: storage layout.
--   **uplo**: specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced.
+-   **uplo**: specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied.
```
This change should be reverted. "should be referenced" is intentional and follows the BLAS docs.

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -183,21 +183,65 @@ console.log( A );
+The function accepts the following arguments:
+
+-   **layout**: `[in] CBLAS_LAYOUT` storage layout.
+-   **uplo**: `[in] CBLAS_UPLO` specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied.
```
Same comment regarding description.

ShabiShett07 committed:
chore: minor clean-up

Co-authored-by: Athan <kgryte@gmail.com>
Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
chore: add suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: add suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: missing_dependencies
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, I have made changes as per your suggestions.

ShabiShett07 committed:
chore: update jsdoc

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -76,13 +76,13 @@ var Float64Array = require( '@stdlib/array/float64' );
 
 // Initial arrays...
 var x0 = new Float64Array( [ 1.0, 1.0, 1.0, 1.0 ] );
```
```suggestion
var x0 = new Float64Array( [ 0.0, 3.0, 2.0, 1.0 ] );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -76,13 +76,13 @@ var Float64Array = require( '@stdlib/array/float64' );
 
 dsyr( 'row-major', 'upper', 3, 1.0, x1, -1, A, 3 );
-// A => <Float64Array>[ 2.0, 3.0, 4.0, 0.0, 2.0, 3.0, 0.0, 0.0, 2.0 ]
+// A => <Float64Array>[ 2.0, 3.0, 4.0, 2.0, 2.0, 3.0, 3.0, 2.0, 2.0 ]
```
```suggestion
// A => <Float64Array>[ 2.0, 4.0, 6.0, 2.0, 5.0, 8.0, 3.0, 2.0, 10.0 ]
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -149,11 +149,17 @@ var opts = {
 var N = 3;
 
-var A = ones( N*N, opts.dtype );
+// Define 3x3 symmetric matrices:
```
```suggestion
// Create N-by-N symmetric matrices:
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -149,11 +149,17 @@ var opts = {
+var A1 = ones( N*N, opts.dtype );
+var A2 = ones( N*N, opts.dtype );
+
 var x = discreteUniform( N, -10.0, 10.0, opts );
```
```suggestion
// Create a random vector:
var x = discreteUniform( N, -10.0, 10.0, opts );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
````diff
@@ -183,21 +189,65 @@ console.log( A );
+-   **LDA**: `[in] CBLAS_INT` stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`).
+
+```c
+void c_dsyr( const CBLAS_LAYOUT layout, const CBLAS_UPLO uplo, const CBLAS_INT N, const double alpha, const double *x, const CBLAS_INT strideX, double *A, const CBLAS_INT LDA )
````
```suggestion
void c_dsyr( const CBLAS_LAYOUT layout, const CBLAS_UPLO uplo, const CBLAS_INT N, const double alpha, const double *X, const CBLAS_INT strideX, double *A, const CBLAS_INT LDA )
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
````diff
@@ -183,21 +189,65 @@ console.log( A );
 
 ```c
-TODO
+void c_dsyr_ndarray( const CBLAS_UPLO uplo, const CBLAS_INT N, const double alpha, const double *x, const CBLAS_INT strideX, const CBLAS_INT offsetX, double *A, const CBLAS_INT strideA1, const CBLAS_INT strideA2, const CBLAS_INT offsetA )
````
```suggestion
void c_dsyr_ndarray( const CBLAS_UPLO uplo, const CBLAS_INT N, const double alpha, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX, double *A, const CBLAS_INT strideA1, const CBLAS_INT strideA2, const CBLAS_INT offsetA )
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -219,7 +269,46 @@ TODO
+        printf( "A1[ %i ] = %f\n", i, A1[ i ] );
+    }
+
+    // Perform the symmetric rank 1 operation `A = α*x*x^T + A` using alternative semantics indexing:
```
```suggestion
    // Perform the symmetric rank 1 operation `A = α*x*x^T + A` using alternative indexing semantics:
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/examples/c/example.c:
```diff
@@ -0,0 +1,58 @@
+		3.0, 2.0, 1.0
+	};
+
+	// Define `x` vector:
```
```suggestion
	// Define a vector:
```

kgryte committed:
docs: update comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/README.md:
```diff
@@ -219,7 +270,46 @@ TODO
+        3.0, 2.0, 1.0
+    };
+
+    // Define `x` vector:
```
```suggestion
    // Define a vector:
```

kgryte committed:
docs: update comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/examples/c/example.c:
```diff
@@ -0,0 +1,58 @@
+		printf( "A1[ %i ] = %f\n", i, A1[ i ] );
+	}
+
+	// Perform the symmetric rank 1 operation `A = α*x*x^T + A` using alternative semantics indexing:
```
```suggestion
	// Perform the symmetric rank 1 operation `A = α*x*x^T + A` using alternative indexing semantics:
```

kgryte committed:
docs: update comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/examples/index.js:
```diff
@@ -28,8 +28,14 @@ var opts = {
+// Define 3x3 symmetric matrices:
+var A1 = ones( N*N, opts.dtype );
+var A2 = ones( N*N, opts.dtype );
+
```
```suggestion
// Create N-by-N symmetric matrices:
var A1 = ones( N*N, opts.dtype );
var A2 = ones( N*N, opts.dtype );

// Create a random vector:
```

kgryte committed:
docs: update comments

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/addon.c:
```diff
@@ -0,0 +1,96 @@
+	if ( layout == CblasColMajor ) {
+		sa1 = 1;
+		sa2 = LDA;
+	} else { // layout === CblasRowMajor
```
```suggestion
	} else { // layout == CblasRowMajor
```

kgryte committed:
docs: update comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr.c:
```diff
@@ -0,0 +1,83 @@
+/**
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param layout    storage layout
```
```suggestion
* @param layout   storage layout
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr.c:
```diff
@@ -0,0 +1,83 @@
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param layout    storage layout
+* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
```
```suggestion
* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
```

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr.c:
```diff
@@ -0,0 +1,83 @@
+	if ( layout == CblasColMajor ) {
+		sa1 = 1;
+		sa2 = LDA;
+	} else { // layout === CblasRowMajor
```
```suggestion
	} else { // layout == CblasRowMajor
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_cblas.c:
```diff
@@ -0,0 +1,37 @@
+/**
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param layout    storage layout
```
```suggestion
* @param layout   storage layout
```

kgryte committed:
docs: fix alignment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_cblas.c:
```diff
@@ -0,0 +1,37 @@
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A` where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param layout   storage layout
+* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
```
```suggestion
* @param uplo     specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_ndarray.c:
```diff
@@ -0,0 +1,118 @@
+/**
+* Performs the symmetric rank 1 operation `A = α*x*x^T + A`, using alternative indexing semantics and where `α` is a scalar, `x` is an `N` element vector, and `A` is an `N` by `N` symmetric matrix.
+*
+* @param uplo      specifies whether the upper or lower triangular part of the symmetric matrix `A` is supplied
```
```suggestion
* @param uplo      specifies whether the upper or lower triangular part of the symmetric matrix `A` should be referenced
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/dsyr/src/dsyr_ndarray.c:
```diff
@@ -0,0 +1,118 @@
+		}
+		return;
+	}
+	// ( isrm && uplo == CblasUpper ) || ( !isrm && uplo === CblasLower )
```
```suggestion
	// ( isrm && uplo == CblasUpper ) || ( !isrm && uplo == CblasLower )
```

kgryte committed:
docs: update comment

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM after clean-up. Will make a follow-up commit adding additional tests.

kgryte commented:
Follow-up commit: https://github.com/stdlib-js/stdlib/commit/c9f4cb9b593f963d25efc20a6bd3b2aeeda515c2
